### DISCUSSION TITLE
Will .CR3 support come to IM?

### DISCUSSION THREAD

**Discussion by ltf3** [ImageMagick/ImageMagick#6077] [Development]
### ImageMagick version

7.1.02

### Operating system, version and so on

MacOS 12.4

### Description

Hi

Is there any sign of Canon  .CR3 RAW support coming to IM? 

I'm not a developer but use software which is dependent on IM. All our Cameras are now exclusively CR3  when shooting RAW ) and unfortunately IM cant process the files to provide previews. 

Perhaps there are licensing hurdles?   Or is it just development resources? I understand IM is open source so that is always a concern.

Any thoughts would be gratefully received.

best

**Comment by snibgo** [comment]
With pre-built IM v7.1.0-49 in Windows, using the builtin delegate libraw, there is no problem reading .CR3 files.

**Comment by ltf3** [comment]
Thanks for the prompt reply!

Unfortunately the developer was a bit cagey about what their platform is running on  ( it is a SaaS product ) but given the size of company I'd say they are based on some kind of Linux. However  I got a sense that the deployed  version of IM might be older which probably  accounts for my issues?

**Comment by snibgo** [comment]
Could be. Or maybe their IM was built with an old version of libraw, or maybe no libraw. If you can, try "`magick -list format`". Is CR3 listed?

**Reply by raleighlittles** [reply]
Are you sure that is the correct command for determining which files are supported?

If I run that with `convert` instead of `magick`, I see CR3 in the list:

```
$ convert -list format | grep CR3
      CR3  DNG       r--   Canon Digital Camera Raw Image Format
```

But when I try to actually convert a CR3 file to JPG, it fails:

```
convert-im6.q16: delegate failed `'ufraw-batch' --silent --create-id=also --out-type=png --out-depth=16 --output='%u.png' '%i'' @ error/delegate.c/InvokeDelegate/1966.
convert-im6.q16: unable to open image `/tmp/magick-JiTHBItCX4dLy-rnDOC8ygADP5gh52AO.ppm': No such file or directory @ error/blob.c/OpenBlob/2924.
convert-im6.q16: no images defined `r7.jpg' @ error/convert.c/ConvertImageCommand/3229.
```

**Comment by snibgo** [comment]
IM delegates the reading of CR3 files, usually to libraw. Older versions of IM used external delegates: dcraw or ufraw-batch.

CR3 is a Canon format. Different Canon cameras use different varieties of CR3. When a new camera is introduced, it may write CR3 files that libraw cannot yet read. Eventually, libraw is updated for the new format, so IM is able to read the files. In addition, some cameras can write different versions of raw files, for example they might have a setting to use an old version of compression that libraw supports, or a new version of compression that libraw does not yet support. See https://www.libraw.org/supported-cameras

A similar situation holds for Nikon NEF files, and probably other manufacturers.

If your camera is not yet supported by libraw, you might ask on their forum at https://www.libraw.org/forum/14

If libraw does support your camera, but IM says it cannot read your CR3 file, you might ask on that forum, or on this IM forum. If you ask here, please zip the CR3 and post it here. 

A similar situation hold for other delegates, dcraw and ufraw-batch.
